File: Dev/C++/Mascaret/src/VEHA/Behavior/BasicActions/PlayAnimationAction.cpp
#include "VEHA/Behavior/BasicActions/PlayAnimationAction.h"
#include "VEHA/Behavior/BasicActions/PlayAnimationBehaviorExecution.h"

namespace VEHA
{

PlayAnimationAction::PlayAnimationAction()
{
	_kind = "PlayAnimation";
}

PlayAnimationAction::~PlayAnimationAction()
{
}

shared_ptr<BehaviorExecution> PlayAnimationAction::createBehaviorExecution(shared_ptr<InstanceSpecification> host, const Parameters& p, bool sync)
{
	shared_ptr<PlayAnimationBehaviorExecution> behavior(new PlayAnimationBehaviorExecution(shared_dynamic_cast<PlayAnimationAction>(shared_from_this()),host,p));
	return behavior;
}

}

File: Dev/CS/Mascaret/Mascaret/VEHA/Behavior/BasicActions/PlayAnimationAction.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Mascaret
{
    class PlayAnimationAction : Action, IEquatable<PlayAnimationAction>
    {

        public string animationName;

        public PlayAnimationAction()
        {
            Kind = "PlayAnimation";
        }

        public bool Equals(PlayAnimationAction other)
        {
            if (other == this)
                return true;
            return false;
        }

        public override BehaviorExecution createBehaviorExecution(InstanceSpecification host, Dictionary<string, ValueSpecification> p, bool sync)
        {
            PlayAnimationBehaviorExecution behavior = new PlayAnimationBehaviorExecution(this, host, p);
            return behavior;
        }

    }
}
